generated/src/aws-cpp-sdk-AWSMigrationHub/source/MigrationHubClient.cpp

```diff
@@ -171,152 +174,390 @@ AssociateCreatedArtifactOutcome MigrationHubClient::AssociateCreatedArtifact(con
 {
   AWS_OPERATION_GUARD(AssociateCreatedArtifact);
   AWS_OPERATION_CHECK_PTR(m_endpointProvider, AssociateCreatedArtifact, CoreErrors, CoreErrors::ENDPOINT_RESOLUTION_FAILURE);
-  ResolveEndpointOutcome endpointResolutionOutcome = m_endpointProvider->ResolveEndpoint(request.GetEndpointContextParams());
-  AWS_OPERATION_CHECK_SUCCESS(endpointResolutionOutcome, AssociateCreatedArtifact, CoreErrors, CoreErrors::ENDPOINT_RESOLUTION_FAILURE, endpointResolutionOutcome.GetError().GetMessage());
-  return AssociateCreatedArtifactOutcome(MakeRequest(request, endpointResolutionOutcome.GetResult(), Aws::Http::HttpMethod::HTTP_POST, Aws::Auth::SIGV4_SIGNER));
+  auto tracer = m_telemetryProvider->getTracer(this->GetServiceClientName(), {});
+  auto span = tracer->CreateSpan(Aws::String(this->GetServiceClientName()) + ".AssociateCreatedArtifact",
+    {{ "rpc.method", request.GetServiceRequestName() }, { "rpc.service", this->GetServiceClientName() }, { "rpc.system", "aws-api" }},
```

Comment:
if `this->GetServiceClientName()` is used to get over-writable service name (that can be updated by customer to update user agent) - then good

otherwise, SERVICE_NAME static variable will serve this purpose.

Reply:
I dont think we will be able to use SERVICE_NAME because we add the service name dimensions in XmlClient/JsonClient/AwsClient where it is not accessible. Also SERVICE_NAME and GetServiceClientName could and do return different values, and they are needed to be the same for dimensions to work correctly.

that function is defined as

```
inline virtual const char* GetServiceClientName() const { return m_serviceName.c_str(); }
```

so i think it is ok to leave as is, let me know what you think though